src/org/labkey/targetedms/TargetedMSServiceImpl.java

```diff
@@ -228,4 +237,10 @@ public Integer importSkylineDocument(ViewBackgroundInfo info, Path skylinePath)
     {
         return TargetedMSManager.addRunToQueue(info, skylinePath);
     }
+
+    @Override
+    public List<? extends ISampleFile> hasData(List<? extends ISampleFile> sampleFiles, Container container)
```

Comment:
I'd expect a "has" method to return true/false. Make this getData() or getSampleFiles() instead?

Reply:
I changed it to getSampleFilesWithData(). Hope that is OK.